### PR TITLE
Starting positions same colour

### PR DESCRIPTION
**Pull request by codeallthethingz**
Added an `getUnoccupiedPointEven` and `getUnoccupiedPointOdd`

Added both methods and then randomly chose one at game creation.  I did this so we wouldn't favour snakes on some squares than others.  Like, snakes that aren't good at avoiding top left would be at a disadvantage if we only used even starting points.

### CODE REVIEW

File: rules/create_test.go [hunk 1]
```diff
@@ -64,14 +64,24 @@ func TestCreateInitialGameWithColour(t *testing.T) {
...
 		Snakes: []*pb.SnakeOptions{
 			{URL: url},
+			{URL: url},
 		},
 	})
 	require.NoError(t, err)
 	require.Len(t, frame, 1)
-	require.Len(t, frame[0].Snakes, 1)
+	require.Len(t, frame[0].Snakes, 2)
+	if isEven(frame[0].Snakes[0].Body[0]) {
```
Comment (codeallthethingz) [inline rules/create_test.go]: If this reverts to random, this test will become flappy.  Can you think of a good way to make it deterministic?

File: rules/create.go [hunk 2]
```diff
@@ -88,13 +89,17 @@ func getTournamentStartPoint(size, index int32, snakes []*pb.Snake) *pb.Point {
 
 func getSnakes(req *pb.CreateRequest) ([]*pb.Snake, error) {
 	var snakes []*pb.Snake
-
+	even := rand.Float32() < 0.5
```
Comment (dlsteuer) [inline rules/create.go]: super minor optimization, but this is only used if it's not a tournament board size, could be calculated inside the else at line 98

Comment (codeallthethingz) [reply rules/create.go]: It needs to be outside the for loop for each snake.
And actually, this optimization is how the behaviour could revert to random which would introduce the flakey test above.

File: rules/create_test.go (hunk 1 above)
Comment (codeallthethingz) [reply rules/create_test.go]: Deterministic seed is out of my comfort zone, do you want to take over this PR or I can create a separate issue for that?

Comment (codeallthethingz) [reply rules/create_test.go]: Another approach would be to
1. Have the engine take an optional parameter of starting food at game creation.
2. Create a large board 100x10 for the test, fill all the squares with food apart from 1 even and 1 odd.
3. assert that 1 remaining even square is odd or even.

If the create reverts to random, this test will now fail 999 times out of 1000 which is a pretty strong test.

Comment (dlsteuer) [reply rules/create_test.go]: By deterministic seed, I just mean using `rand.Seed(<some number>)` this way you know the order of randomness from all the call in the method you are testing.

File: rules/create.go (hunk 2 above)
Comment (dlsteuer) [reply rules/create.go]: oh yeah you are correct.

File: rules/create_test.go (hunk 1 above)
Comment (codeallthethingz) [reply rules/create_test.go]: ok, so action I'm gonna take tonight
Put a comment on the test function about never running this in parallel.
Add the `rand.seed` call to the test function.

File: rules/create_test.go [hunk 3]
```diff
@@ -54,7 +55,9 @@ func TestCreateInitialGame_MoreSnakesThanSpace(t *testing.T) {
 	require.Error(t, err)
 }
 
+// Do not run this test in paralell as it sets the random seed.
```
Comment (dlsteuer) [inline rules/create_test.go]:
```suggestion
// Do not run this test in parallel as it sets the random seed.
```

**Pull request merged**